### DISCUSSION TITLE
How is the GLSP release strategy?

### DISCUSSION THREAD

**Discussion by ivy-lli** [eclipse-glsp/glsp#797] [General] [answered]
Hi all!

Sorry if I missed it, but I lately discovered this blog entry on your website, about the theia community release: https://eclipsesource.com/blogs/2022/10/18/the-eclipse-theia-community-release-2022-09/ 

So my question is, does GLSP tries to follow this release strategy?
Which would mean:

- 1 release every 3 months
- 2nd month RC01 release
- 3nd month RC02 release

Would this also mean that every integration (vscode, theia, eclipse, standalone) will get this release or only those how have changes?
And in the end when do you plan to release v1.1.0?

Thanks for your work 🙂

**Comment by tortmayr** [comment, marked as answer]
Hi @ivy-lli,
we have discussed this internally but at the moment following a 3 months release cycle for all GLSP components doesn't make sense for us.
In many components the feature set does not change enough within three months to justify a new release.

We are currently working on implementing the following release strategy.
- Monthly release candidates for all components. We have are not set on a fixed time slot but it will be something within the last week of a month or the first week of the following month (e.g. last friday of the month). We will start this release candidate cycle with the end of November.
- Special treatment for the theia-integration. In case of a new community release we check whether the current theia integration release is still compatible. If not we will publish a patched version just for the theia-integration. The `theia-cr` tag always points to the version that is compatible with the latest community release

We won't follow a fixed schedule for actual releases but the goal is to release a new version at least once a year.
As for a 1.1.0 release we are not set on a fixed date yet as but it will be scheduled for the first quarter of 2023.
